tests/Aspire.Elastic.Clients.Elasticsearch.Tests/AspireElasticClientExtensionsTest.cs

```diff
@@ -0,0 +1,146 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Elastic.Clients.Elasticsearch;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+
+namespace Aspire.Elastic.Clients.Elasticsearch.Tests;
+
+public class AspireElasticClientExtensionsTest : IClassFixture<ElasticsearchContainerFixture>
+{
+
+    private const string DefaultConnectionName = "elasticsearch";
+
+    private readonly ElasticsearchContainerFixture _containerFixture;
+
+    public AspireElasticClientExtensionsTest(ElasticsearchContainerFixture containerFixture)
+    {
+        _containerFixture = containerFixture;
+    }
+    private string DefaultConnectionString => _containerFixture.GetConnectionString();
+
+    [Fact]
+    [RequiresDocker]
+    public void AddKeyedElasticsearchClient_HealthCheckShouldNotBeRegisteredWhenDisabled()
+    {
+        var builder = CreateBuilder(DefaultConnectionString);
+
+        builder.AddKeyedElasticsearchClient(DefaultConnectionName, settings =>
+        {
+            settings.DisableHealthChecks = true;
+        });
+
+        using var host = builder.Build();
+
+        var healthCheckService = host.Services.GetService<HealthCheckService>();
+
+        Assert.Null(healthCheckService);
+
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task AddElasticsearchClient_HealthCheckShouldBeRegisteredWhenEnabled()
+    {
+        var builder = CreateBuilder(DefaultConnectionString);
+
+        builder.AddElasticsearchClient(DefaultConnectionName, settings =>
+        {
+            settings.DisableHealthChecks = false;
+            settings.HealthCheckTimeout = 1;
+        });
+
+        using var host = builder.Build();
+
+        var healthCheckService = host.Services.GetRequiredService<HealthCheckService>();
+
+        var healthCheckReport = await healthCheckService.CheckHealthAsync();
+
+        var healthCheckName = "Elastic.Clients.Elasticsearch";
+
+        Assert.Contains(healthCheckReport.Entries, x => x.Key == healthCheckName);
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task AddKeyedElasticsearchClient_HealthCheckShouldBeRegisteredWhenEnabled()
```

Comment:
The keyed, and non-keyed versions of the tests can be combined like:
https://github.com/dotnet/aspire/blob/3406514607bb775cb5c9693c269832ca5c5a4170/tests/Aspire.NATS.Net.Tests/AspireNatsClientExtensionsTests.cs#L121-L126